eng/Versions.props

```diff
@@ -27,6 +27,7 @@
     <DefaultNetFxTargetFramework>net472</DefaultNetFxTargetFramework>
     <DefaultNetCoreTargetFramework>net8.0</DefaultNetCoreTargetFramework>
     <DefaultNetCoreTargetFrameworks>$(DefaultNetCoreTargetFramework);net7.0</DefaultNetCoreTargetFrameworks>
+    <DefaultNetCoreTargetFrameworks Condition="'$(DotNetBuildFromSource)' == 'true'">$(DefaultNetCoreTargetFramework)</DefaultNetCoreTargetFrameworks>
```

Comment:
Isn't this going to break when it merges into dotnet/sdk for 7.0 releases? Basically doesn't this have the exact same problem roslyn had that forced us to multi-target `net7.0;net8.0` when `'$(DotNetBuildFromSourceFlavor)' != 'Product'`?

Reply:
Yes, I overlooked that this flows into multiple releases.  Thanks for the catch.  I will open another PR.